### PR TITLE
refactor(settings): add style for buttons instead of links

### LINKED ISSUE #624: Use buttons instead of links
From https://github.com/mozilla/fxa-content-server/pull/6239/files#r192063057 and https://github.com/mozilla/fxa-bugzilla-mirror/issues/504, there some places where we use links that do not navigate to anywhere. In these scenarios, we should probably be using a button.

Ideally, we want the buttons to be styled like this
screen shot 2018-06-05 at 11 49 35 am

where there is an image and the text is centered below it.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-755)
┆Priority: Medium
┆Issue Number: FXA-755

### PR DESCRIPTION
**Pull request by AlbionaHoti**
@shane-tomlinson @lmorchard Can you please review this :) 

fixes #624

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/templates/settings/avatar_change.mustache [hunk 1]
```diff
@@ -12,10 +12,10 @@
 
     <nav id="avatar-options">
       <input type="file" id="imageLoader" name="imageLoader" accept="image/png, image/jpeg"/>
-      <a href="#" id="file">{{#t}}Upload{{/t}}</a>
+      <button class="save-option-button" id="file">{{#t}}Upload{{/t}}</button>
```
[the remark this answers is not in the document]
Comment (AlbionaHoti) [reply packages/fxa-content-server/app/scripts/templates/settings/avatar_change.mustache]: I added the save-option-button class for some reasons:

Using buttons -
1. The recovery codes
2. The avatar part

After replacing the <a> tags with <button> - the style was changed to this:
one
one2

1. With the save-option-button class, the view returned as it was - but now I just figured it out that I can use the save-option class only to make the view look as it was with buttons too.
         1.1 I will add some changes in save-option so the design would remain the same when using buttons.

2. As for the hover on the recovery codes - I will add the hover there too.

3. Now I will use save-option class in recovery_codes and recovery_key files.
4. I will use save-option-button in avatar_change only.

[the remark this answers is not in the document]
Comment (AlbionaHoti) [reply packages/fxa-content-server/app/scripts/templates/settings/avatar_change.mustache]: @shane-tomlinson I just add the changes, can you review them?

1. The hover is working in recovery keys buttons too
2. I removed the save-option-button from recovery_codes and recovery_key files
    2.1 Add change to save-options class so the button is good now 
3. Only the avatar_change file is using the save-option-button class now

**Comment by shane-tomlinson** [comment]
@AlbionaHoti - this looks much better! The behaviors look better, and there is further room for improvement.

I noticed that a lot of button related CSS is being repeated which is a smell, it'd be great to avoid duplication if at all possible. I suspect this duplication is because the markup used for the buttons in avatar_change.mustache is different to the markup used in recovery_codes.mustache or recovery_key.mustache.

The 2nd two use a button of the form:

```html
                <button class="save-option download-option">
                    <div class="graphic graphic-download-option"></div>
                    <p class="name">{{#t}}Download{{/t}}</p>
                </button>
```
which is surrounded by a 
```html
<div class="button-row save-options">
```

The avatar_change.mustache file uses this:

```html
      <button class="save-option-button" id="file">{{#t}}Upload{{/t}}</button>
```

surrounded by:

```html
    <nav id="avatar-options">
```

I have used your PR here as a basis to do a bit of mucking around to try to simplify. I was able to make avatar_change.mustache use the same button and button-row markup as the recovery_* templates and the result is I was able to simplify the CSS quite a bit.

I can paste my diff, though you might want to have a stab at attempting the change yourself, I don't want to steal your thunder.

A small hint, I moved the `.save-options` CSS out of it's current location into _button-row.scss so that it could be shared more easily.

As you are working through this, one of the behvchanges that need to be done is that if the user hovers over any part of the button, text included, the hover state should be invoked. Currently the hover state only takes effect if I hover over the icon.

**Comment by AlbionaHoti** [comment]
> @AlbionaHoti - this looks much better! The behaviors look better, and there is further room for improvement.
> 
> I noticed that a lot of button related CSS is being repeated which is a smell, it'd be great to avoid duplication if at all possible. I suspect this duplication is because the markup used for the buttons in avatar_change.mustache is different to the markup used in recovery_codes.mustache or recovery_key.mustache.
> 
> The 2nd two use a button of the form:
> 
> ```
>                 <button class="save-option download-option">
>                     <div class="graphic graphic-download-option"></div>
>                     <p class="name">{{#t}}Download{{/t}}</p>
>                 </button>
> ```
> 
> which is surrounded by a
> 
> ```
> <div class="button-row save-options">
> ```
> 
> The avatar_change.mustache file uses this:
> 
> ```
>       <button class="save-option-button" id="file">{{#t}}Upload{{/t}}</button>
> ```
> 
> surrounded by:
> 
> ```
>     <nav id="avatar-options">
> ```
> 
> I have used your PR here as a basis to do a bit of mucking around to try to simplify. I was able to make avatar_change.mustache use the same button and button-row markup as the recovery_* templates and the result is I was able to simplify the CSS quite a bit.
> 
> I can paste my diff, though you might want to have a stab at attempting the change yourself, I don't want to steal your thunder.

**Thank you for your comments. I add the changes and use the save-options in three files the `recovery_*` templates and in the avatar_change template, modified it and here is the result.**

> 
> A small hint, I moved the `.save-options` CSS out of its current location into _button-row.scss so that it could be shared more easily.
> 
**I moved the `.save-options` class into _button_row.scss**
> As you are working through this, one of the best changes that need to be done is that if the user hovers over any part of the button, text included, the hover state should be invoked. Currently, the hover state only takes effect if I hover over the icon.

**I included the hover for the `.name` too**

**One thing that I am not sure how to solve is the border when you click on the camera, is removed I think from the <a> tag and when you right click on it the border is shown**
@shane-tomlinson

**Comment by AlbionaHoti** [comment]
@shane-tomlinson Can you review the last change, please :)

**Comment by farhan787** [comment]
@AlbionaHoti you need to sign this commit and fix the conflicting files.

File: packages/fxa-content-server/app/scripts/templates/settings/avatar_change.mustache [hunk 2]
```diff
@@ -10,14 +10,23 @@
...
-      <a href="/settings/avatar/camera" id="camera">{{#t}}Camera{{/t}}</a>
+    <input type="file" id="imageLoader" name="imageLoader" accept="image/png, image/jpeg"/>
+    <div class="button-row save-options">
+      
+      <button class="save-option" id="file">
+          <div class="graphic graphic-upload-option"></div>
+          <p class="name">{{#t}}Upload{{/t}}</p>
+      </button>
+      <button class="save-option" id="camera">
+          <a href="/settings/avatar/camera">
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/app/scripts/templates/settings/avatar_change.mustache]: Wrapping the anchor here in the button makes the link inaccessible using keyboard. Can you unwrap just this link?

Comment (AlbionaHoti) [reply packages/fxa-content-server/app/scripts/templates/settings/avatar_change.mustache]: I will add the changes this week. I am interested to finish this task :) 

Thank you for your feedback, I appreciate it.

**Comment by shane-tomlinson** [comment]
Hi @AlbionaHoti, thank you for the patch, it would be a shame to let this bit rot and not be merged. Are you still interested in getting this landed? If so, let us know before Oct 1 whether you want to make the necessary updates. If we don't hear from you, we'll either close this or someone within the team will try to get it across the line.

**Comment by AlbionaHoti** [comment]
> Hi @AlbionaHoti, thank you for the patch, it would be a shame to let this bit rot and not be merged. Are you still interested in getting this landed? If so, let us know before Oct 1 whether you want to make the necessary updates. If we don't hear from you, we'll either close this or someone within the team will try to get it across the line.

Hi @shane-tomlinson, thank you for your feedback and comment, I will continue now to add the changes and land this issue, I hope until the 1st of October it will be all done. Will let you know if I need anything.

**Comment by vladikoff** [comment]
Please reopen a rebased PR for this if possible! Thanks!

**Pull request closed without merging**